IpmiFeaturePkg/BmcAcpiPowerState/BmcAcpiPowerStateSmm.c

```diff
@@ -0,0 +1,185 @@
+/** @file
+    BmcAcpiPowerStateSmm.c
+
+    Copyright (c) Microsoft Corporation. All rights reserved.
+**/
+
+#include <PiSmm.h>
+#include <Library/SmmServicesTableLib.h>
+#include <Library/DebugLib.h>
+#include <Library/IpmiBaseLib.h>
+#include <IndustryStandard/IpmiNetFnApp.h>
+
+#include <Protocol/SmmExitBootServices.h>
+#include <Protocol/SmmSxDispatch2.h>
+
+/**
+  Send IPMI command to set ACPI power state to BMC.
+
+  @param[in]  SystemPowerState        System Power State enumeration
+  @param[in]  DevicePowerState        Device Power State enumeration
+
+  @retval EFI_SUCCESS     Set ACPI power state to BMC successfully.
+  @retval Other           Failed to set ACPI power state to BMC.
+**/
+EFI_STATUS
+SetBmcAcpiPowerState (
+  IN  UINT8  SystemPowerState,
+  IN  UINT8  DevicePowerState
+  )
+{
+  EFI_STATUS                         Status;
+  IPMI_SET_ACPI_POWER_STATE_REQUEST  SetAcpiPowerState;
+  UINT8                              CompletionCode;
+  UINT32                             DataSize;
+
+  SetAcpiPowerState.SystemPowerState.Bits.StateChange = 1;

```

Comment:
Probably worth zeroing SetAcpiPowerState, event if all fields are set. e.g. ZeroMem(&SetAcpiPowerState, sizeof(SetAcpiPowerState);

Reply:
Added code to initialize SetAcpiPowerState in commit ba9c36e47bd98b1c9a33fcc9b64c018135455ac0. Thank you for your suggetion.